src/bg_cache_worker.ts

```diff
@@ -176,15 +176,40 @@ async function updateCache() {
 
     console.log("Done with executed proposals, now looking for Redeemer redemptions");
 
-    const redeemerInstance = await Arc.RedeemerFactory.deployed();
-    const redeemEventWatcher = redeemerInstance.RedeemerRedeem({ }, { fromBlock: lastCachedBlock, toBlock: latestBlock });
-    const getRedeemEvents = promisify(redeemEventWatcher.get.bind(redeemEventWatcher, -1));
-    const redeemEvents: Array<Arc.DecodedLogEntryEvent<Arc.RedeemerRedeemEventResult>> = await getRedeemEvents();
-    for (let index = 0; index < redeemEvents.length; index++) {
-      const event = redeemEvents[index];
-      console.log("Proposal redeemed through Redeemer", event);
-      await store.dispatch(arcActions.onRedeemEvent(event.args._proposalId));
-    }
+    const redeemEth = await contributionRewardInstance.RedeemEther({}, {fromBlock: lastCachedBlock, toBlock: latestBlock}).get(undefined, -1);
+    redeemEth.forEach(({args: {_proposalId, _avatar, _beneficiary}}) =>
```

Comment:
i don't think we can use foreach because im pretty sure we want to await each dispatch and carry out each event one by one in the cache script. so need to use a regular for loop

Reply:
right, also I noticed that there are some places we can use `Promise.all` instead of a `for` loop with an `await` to make it parallel (e.g. in `getDAOs`). I think it's also the case here.

Reply:
yeah thats true

Reply:
https://github.com/daostack/alchemy/pull/155/commits/859188fae85d97f760f2cc184f8a987fb1f4eb97